clients/src/main/java/org/apache/kafka/clients/InFlightRequests.java

```diff
@@ -26,13 +26,19 @@
 import java.util.concurrent.atomic.AtomicInteger;
 
 /**
- * The set of requests which have been sent or are being sent but haven't yet received a response
+ * The set of requests which have been sent, or are being sent, but have not yet
+ * received a response. This class is not thread-safe as it pertains to
```

Comment:
Let's use a new paragraph for the thread-safety description. Also, what do we mean by `relatively accurate`? I think that's a bit ambiguous. Maybe call it `eventually consistent` or something like that?